app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,32 @@
+class OrdersController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_order, only: [:show]
+
+  def new
+    @order = Order.new
+  end
+
+  def create
+    @order = Order.new(user: current_user, total: current_user.cart.total_cart_price)
+    if @order.save
+      @order.purchase_line_items
+      redirect_to order_path(@order), notice: "Your order has been completed"
```

Comment:
You are referencing the `show` action here and elsewhere, but I don't see a `show` method here. Is there an implicit `show` method? Since the controller is not under test, I'm not sure if this method would throw an error.